### ISSUE TITLE
Newtonsoft Json assembly manifest mismatch?

### ISSUE THREAD

**Issue by DeathTBO** [aspnet-contrib/AspNet.Security.OpenId.Providers#104]
Using Providers version 3.1.0, and DotNet Core 3.1

I'm developing an Electron.Net application to pair with a website. I've already implemented OpenID and Steam login with no issues on the Asp.Net Core website, but I'm running into issues on the Electron client. I started by copying the the implementation for the website (which is rather straightforward). The one difference was I had to turn NodeIntegration off.

After logging into Steam, upon redirect an error pops up. Newtonsoft.Json is automatically installed to Version 10.0.3 via Nuget. I've also tried other version, but none seemed to work. I actually thought Newtonsoft was replaced by System.Text.Json with the latest version.

Output:

```
      Error from RemoteAuthentication: Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The located assembly's manifest definition does not match the assembly reference. (0x80131040).
stdout: fail: Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware[1]
      An unhandled exception has occurred while executing the request.
System.Exception: An error was encountered while handling the remote login.
 ---> System.IO.FileLoadException: Could not load file or assembly 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)
File name: 'Newtonsoft.Json, Version=10.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'
   at AspNet.Security.OpenId.OpenIdAuthenticatedContext..ctor(HttpContext context, AuthenticationScheme scheme, OpenIdAuthenticationOptions options, AuthenticationTicket ticket)
   at AspNet.Security.OpenId.Steam.SteamAuthenticationHandler.<>c__DisplayClass1_0.<<CreateTicketAsync>g__RunAuthenticatedEventAsync|0>d.MoveNext() in /_/src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs:line 119
--- End of stack trace from previous location where exception was thrown ---
   at AspNet.Security.OpenId.Steam.SteamAuthenticationHandler.CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, String identifier, IReadOnlyDictionary`2 attributes) in /_/src/AspNet.Security.OpenId.Steam/SteamAuthenticationHandler.cs:line 56
   at AspNet.Security.OpenId.OpenIdAuthenticationHandler`1.HandleRemoteAuthenticateAsync() in /_/src/AspNet.Security.OpenId/OpenIdAuthenticationHandler.cs:line 211
   at Microsoft.AspNetCore.Authentication.RemoteAuthenticationHandler`1.HandleRequestAsync()
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Authentication.RemoteAuthenticationHandler`1.HandleRequestAsync()
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)

```

**Comment by martincostello** [comment]
> I actually thought Newtonsoft was replaced by System.Text.Json with the latest version.

Newtonsoft.Json has been removed from our 5.0.0 branch for the next major release, but it is still referenced as part of the 3.x packages shipped from this repo. Removing it between 3.0 and 3.1 would have been a breaking change due to `JObject` being in the public API surface area, so it was just deprecated as part of 3.1.

https://github.com/aspnet-contrib/AspNet.Security.OpenId.Providers/blob/fe99a4371f222c40c3a5b7351f0fd12ff7e0094c/src/AspNet.Security.OpenId/Events/OpenIdAuthenticatedContext.cs#L62-L63

I'm not sure why you'd be getting that error though. Is something in your build process overwriting the assembly with an incompatible version maybe?

**Comment by DeathTBO** [comment]
Rather strange. I decided to try publishing a release build (electronize build), and it worked as expected. The electron system first builds into a normal release, then builds that into an AppImage file (I'm on Linux). However using the electronize start method does not work. The file structure for the AppImage build is completely different than the release/debug and electronize start.

I ran checksums on the different builds and the actual Newtonsoft.Json 10.0.3 dll's. They all matched except for the final AppImage build. Which was a whopping 1.7mb compared to the other ~624kb.

I don't think this is an issue with the OpenID library. Looking forward to the 5.0.0 release :P
